src/main/java/com/google/devtools/build/lib/remote/blobstore/http/HttpBlobStore.java

```diff
@@ -267,6 +268,8 @@ private Channel acquireUploadChannel() throws InterruptedException {
                 p.addLast(new HttpObjectAggregator(10 * 1024));
                 p.addLast(new HttpRequestEncoder());
                 p.addLast(new ChunkedWriteHandler());
+                p.addFirst("read-timeout-handler", new ReadTimeoutHandler(timeoutSeconds));
```

Comment:
The semantics of `ReadTimeoutHandler` are such that it puts a total limit of `timeoutSeconds` on an upload as it will trigger when not having received any data for `timeoutSeconds`. So I believe this is not what we want here, as we do want timeouts to be progressive (i.e. timeout if no bytes written for N seconds.)

Reply:
I was actually happy with limiting the total time for any upload or download to X seconds.  If the timeout for the remote cache is set to 5 seconds, I don't want any upload or download to take longer than 5 seconds.

Reply:
I understand. The issue with that is that it sets an upper bound on the maximum output size that can be downloaded on a given machine. In this remote code we are trying to get into a world where the default value will be set to `--remote_timeout=1` or so, meaning that we'll timeout if no data has been received for 1 second and users essentially don't have to touch this flag anymore. Makes sense?

Reply:
I changed the code to use idle timeouts, so I believe this problem should be fixed.